helix-term/src/ui/document.rs

```diff
@@ -58,13 +74,22 @@ impl<H: Iterator<Item = HighlightEvent>> Iterator for StyleIter<'_, H> {
                     if start == end {
                         continue;
                     }
+                    let mut is_selection = false;
                     let style = self
                         .active_highlights
                         .iter()
                         .fold(self.text_style, |acc, span| {
+                            // find scoped used in the selection to later determine if text is in a selection
+                            if self.theme.selection_scopes().binary_search(&span.0).is_ok() {
```

Comment:
This is pretty hacky and would be prone tincause problems if we reuse theme scopes somewhere.  This also basically hardcoded yet another function in the rendering code. One of my big goals with #5420 was to avoid exactly that problem. 


Yo that end the `TextDecoration` API exists. It's currently not quite fleshed out enough but once #6417 lands the API will be expanded/refactored so you can draw decorations as specific char anchors. Virtual text is special cased here because this is genuinely the only place we know whether a grapheme is virtual text (since it's nitnpartnof the normakd document) but selections are a high keven concept that the rendering shouldn't be aware of

Reply:
So should I wait merge of your PRs or overhaul mine? Tbh, if it's better to wait, I guess I'll close it for now, because I'm not sure if I'll have time in six months or a year to maintain it. Hope you are not tired of my attempts :smiley:

Reply:
no worries. Unfortunately, I do think it would be better to wait here. I hope you understand. In this case if we were to merge your PR as is I would have to undo the entire change in the future (and then reimplement it with the new API). Thanks for contributing tough!

Reply:
np, I understand